cmake/FindQuickLintJsEmacs.cmake

```diff
@@ -0,0 +1,81 @@
+# Copyright (C) 2020  Matthew "strager" Glazar
+# See end of file for extended copyright information.
+
+if (QUICK_LINT_JS_EMACS)
+  set(QUICK_LINT_JS_EMACS "${QUICK_LINT_JS_EMACS}")
+else ()
+  find_program(QUICK_LINT_JS_EMACS "emacs")
+endif ()
+
+if (NOT QUICK_LINT_JS_EMACS)
+  message(WARNING "Emacs not found. Skipping... ")
+  return ()
+endif ()
+
+execute_process(
+  COMMAND
+    ${QUICK_LINT_JS_EMACS}
+    -Q -batch
+    --eval "(princ (format \"%s.%s\" emacs-major-version emacs-minor-version))"
+  RESULT_VARIABLE EMACS_EXIT_CODE
+  OUTPUT_VARIABLE EMACS_VERSION)
+
+if (NOT EMACS_EXIT_CODE EQUAL 0)
+  message(WARNING "Emacs (${QUICK_LINT_JS_EMACS}) found but can't get its version. Skipping...")
+  return ()
+endif()
+
+if (NOT EMACS_VERSION GREATER_EQUAL 24.5)
+  message(WARNING "Emacs found (${QUICK_LINT_JS_EMACS}), but version ${EMACS_VERSION} is not supported. Skipping...")
+  return ()
+endif ()
+
+set(QUICK_LINT_JS_EMACS_FOUND TRUE)
+message(STATUS "Found Emacs: (${QUICK_LINT_JS_EMACS}) suitable version ${EMACS_VERSION} minimum required is 24.5")
+
+macro(emacs_pkg_target NAME FILE)
+  cmake_parse_arguments(
+    ""
+    ""
+    "OUTPUT"
+    ""
+    ${ARGN})
+
+  add_custom_command(
+    OUTPUT ${_OUTPUT}
+    COMMAND
+      ${QUICK_LINT_JS_EMACS}
+      -Q -batch
+      -l package --eval "(package-initialize)"
+      --eval "(add-to-list 'package-directory-list \"${CMAKE_CACHEFILE_DIR}\")"
```

Comment:
Why add CMAKE_CACHEFILE_DIR? What files are in the build dir that we need to expose to Emacs?

Reply:
that's the output directory for the package, otherwise Emacs would make it somewhere on its defaults, (probably `~/.emacs.d/elpa`), i used the CMAKE_CACHEFILE_DIR because i assume cmake calls `${QUICK_LINT_JS_EMACS}` with current directory being the same from it was invoked from, so the root of build directory would have been filled with emacs package folders instead of `${BUILD_DIR}/plugins/emacs`. I cloud've made this from emacs by appending `plugins/emacs` to cwd, or you have something else in mind?

Reply:
> that's the output directory for the package

Oh. It looked like an input directory to me.

`${CMAKE_CURRENT_BINARY_DIR}` might be better.

> I cloud've made this from emacs by appending `plugins/emacs` to cwd, or you have something else in mind?

I think specifying cwd explicitly is what we should do:
```cmake
  add_custom_command(
    OUTPUT ${_OUTPUT}
    COMMAND
      ${QUICK_LINT_JS_EMACS}
      -Q -batch
      -l package --eval "(package-initialize)"
      --eval "(add-to-list 'package-directory-list .)"
      ...
    WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"
    ...
  )
```